### PR TITLE
chore: add semantic conventions to phoenix

### PR DESCRIPTION
**Pull request by yordis**

### CODE REVIEW

File: instrumentation/opentelemetry_phoenix/test/opentelemetry_phoenix_test.exs [hunk 1]
```diff
@@ -53,16 +53,17 @@ defmodule OpentelemetryPhoenixTest do
...
              "http.status_code": 200,
              "http.target": "/users/123",
-             "http.user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:81.0) Gecko/20100101 Firefox/81.0",
-             "net.host.ip": "10.211.55.2",
+             "http.user_agent":
+               "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:81.0) Gecko/20100101 Firefox/81.0",
+             "net.sock.host.addr": "10.211.55.2",
              "net.host.port": 4000,
-             "net.peer.ip": "10.211.55.2",
+             "net.sock.peer.addr": "10.211.55.2",
```
Comment (yordis) [inline instrumentation/opentelemetry_phoenix/test/opentelemetry_phoenix_test.exs:66]: https://github.com/open-telemetry/opentelemetry-specification/blob/120f60ff70ed2a64af26017c8fe611f85e09d13f/CHANGELOG.md?plain=1#LL263C77-L263C95

File: instrumentation/opentelemetry_phoenix/test/opentelemetry_phoenix_test.exs [hunk 2]
```diff
@@ -53,16 +53,16 @@ defmodule OpentelemetryPhoenixTest do
     assert %{
              "http.client_ip": "10.211.55.2",
              "http.flavor": :"1.1",
-             "http.host": "localhost",
+             "net.host.name": "localhost",
```
Comment (yordis) [inline instrumentation/opentelemetry_phoenix/test/opentelemetry_phoenix_test.exs:56]: Double-check here, I am not sure if this is correct or not but `http.host` does not exists in the macros

Comment (tsloughter) [reply instrumentation/opentelemetry_phoenix/test/opentelemetry_phoenix_test.exs:56]: Yea, pretty sure this is right.

File: instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex [hunk 3]
```diff
@@ -22,11 +22,13 @@ defmodule OpentelemetryPhoenix do
       end
 
   """
-
-  require OpenTelemetry.Tracer
+  alias OpenTelemetry.SemanticConventions.Trace
```
Comment (tsloughter) [inline instrumentation/opentelemetry_phoenix/lib/opentelemetry_phoenix.ex:25]: I think having it be `SemanticConventions.Trace` would make the attribute keys more clearly convey what they are, like `SemanticConventions.Trace.http_route() => meta.route()`.

`Trace` and `Tracer` are just so similar :). 

But I'm not hardset on it, so would be fine with merging as is as well.

File: instrumentation/opentelemetry_phoenix/test/opentelemetry_phoenix_test.exs (hunk 1 above)
Comment (yordis) [reply instrumentation/opentelemetry_phoenix/test/opentelemetry_phoenix_test.exs:65]: Check here, please!

Comment (tsloughter) [reply instrumentation/opentelemetry_phoenix/test/opentelemetry_phoenix_test.exs:65]: looks right

**Review by tsloughter — APPROVED** [review]

**Pull request merged**